e2e/events/events_test.go

```diff
@@ -0,0 +1,127 @@
+package events
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/e2e/e2eutil"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+// TestEventStream_Ports tests that removing the mapstructure tags from the Port
+// struct results in Port values being serialized correctly to the event stream.
+func TestEventStream_Ports(t *testing.T) {
```

Comment:
This test doesn't really cover the whole change, does it? It looks like we're testing that we can decode the serialized value for the ports, but not that we haven't changed the API. (Similar to my comment above for the `alloc_endpoint_test.go`)

Also, having a new nightly E2E test for this bug seems excessive; E2E tests really need to bring a lot of value to the party. Could we make these fields appear in the existing events E2E test?

Reply:
oh yeah. I could probably find a way to cover this in an existing test.

Reply:
I added `network` stanzas into existing e2e job files.